### PR TITLE
add section "Manipulating queued dataset events through REST API"

### PR DESCRIPTION
**Pull request by Lee-W**
Add docs to illustrate how https://github.com/apache/airflow/pull/37176 can be used.

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst [hunk 1]
```diff
@@ -321,6 +321,33 @@ Example:
 Note that this example is using `(.values() | first | first) <https://jinja.palletsprojects.com/en/3.1.x/templates/#jinja-filters.first>`_ to fetch the first of one dataset given to the DAG, and the first of one DatasetEvent for that dataset. An implementation can be quite complex if you have multiple datasets, potentially with multiple DatasetEvents.
 
 
+Manipulating queued dataset events through REST API
+---------------------------------------------------
+
+.. versionadded:: 2.9
+
+In this example, the DAG "waiting_for_dataset_1_and_2" will be triggered when tasks update both datasets "dataset-1" and "dataset-2". Once "dataset-1" is updated, Airflow creates a record. This ensures that Airflow knows to trigger the DAG when "dataset-2" is updated. We call such records queued dataset events.
```
Comment (kaxil) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst]:
```suggestion
In this example, the DAG ``waiting_for_dataset_1_and_2`` will be triggered when tasks update both datasets "dataset-1" and "dataset-2". Once "dataset-1" is updated, Airflow creates a record. This ensures that Airflow knows to trigger the DAG when "dataset-2" is updated. We call such records queued dataset events.
```

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
cc @fritz-astronomer

**Review by fritz-astronomer — APPROVED** [review]

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst [hunk 2]
```diff
@@ -321,6 +321,34 @@ Example:
...
+``quededEvent`` API endpoints are introduced to manipulate such records.
+
+* Get a queued Dataset event for a DAG: ``/datasets/queuedEvent/{uri}``
+* Get queued Dataset events for a DAG: ``/dags/{dag_id}/datasets/queuedEvent``
+* Delete a queued Dataset event for a DAG: ``/datasets/queuedEvent/{uri}``
+* Delete queued Dataset events for a DAG: ``/dags/{dag_id}/datasets/queuedEvent``
+* Get queued Dataset events for a Dataset: ``/dags/{dag_id}/datasets/queuedEvent/{uri}``
+* Delete queued Dataset events for a Dataset: ``DELETE /dags/{dag_id}/datasets/queuedEvent/{uri}``
+
+ For how to use REST API and the parameters needed for these endpoints, please refer to :doc:`Airflow API </stable-rest-api-ref>`
```
Comment (eladkal) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst:350]:
```suggestion
To read more about the REST API and the parameters needed for these endpoints, please refer to :doc:`Airflow API </stable-rest-api-ref>`
```

File: docs/apache-airflow/authoring-and-scheduling/datasets.rst [hunk 3]
```diff
@@ -321,6 +321,34 @@ Example:
...
+In this example, the DAG ``waiting_for_dataset_1_and_2`` will be triggered when tasks update both datasets "dataset-1" and "dataset-2". Once "dataset-1" is updated, Airflow creates a record. This ensures that Airflow knows to trigger the DAG when "dataset-2" is updated. We call such records queued dataset events.
+
+.. code-block:: python
+
+    with DAG(
+        dag_id="waiting_for_dataset_1_and_2",
+        schedule=[Dataset("dataset-1"), Dataset("dataset-2")],
+        ...,
+    ):
+        ...
```
Comment (eladkal) [inline docs/apache-airflow/authoring-and-scheduling/datasets.rst:338]: What is the added value of this paragraph?
It feels like repeating what is previously explain in Multiple Datasets sectionץ

I think the section should be just about the API endpoints (BTW is it just API or also CLI?)

Comment (Lee-W) [reply docs/apache-airflow/authoring-and-scheduling/datasets.rst:338]: I'm trying to explain the term "queued dataset events" here. Under the hook, it's DatasetDagRunQueue, but ASAIK, we don't want to expose this internal thing

**Pull request merged**